### ISSUE TITLE
Pipe multiple adatper trimming steps is unsupported when using the python API.

### ISSUE THREAD

**Issue by y9c** [marcelm/cutadapt#777]
When using the API code in cutadapt, multiple adapter trimming steps will cause error. 

For example given a NGS sequencing library with the following scheme:

`(p5 adapter)-(UMI)-(inner adapter)-(insert sequence)-(p7 adapter)`

We need to cut `p5 adapter` by `-a`, then cut UMI sequence by `-u`, then cut inner adapter again by `-a`. Since `-a` is triggered twice, cutadapter will raise an error. 

https://github.com/marcelm/cutadapt/blob/33df54550e9d789e03fb9e10a98d18a2a8d71b94/src/cutadapt/report.py#L180

**Comment by y9c** [comment]
Thank you @marcelm for helping with this issue. I wrote a simple tool, cutseq, to simplify the adapter trimming steps. It heavily uses the API of cutadapt, and might solve some of the pain points in issue #606.

**Comment by y9c** [comment]
Thank you very much for your help. I am testing this idea with some real-world data these days, trying to find out the limitations of this API. It meets the requirements of most of the NGS data I have analyzed. It would be great if cutadapt could support this out of the box. I tried to add one new argument for this, but it seems that most of the existing arguments are conflicting with it. Is it okay to suppress all other arguments once the new API is given, or to add a subcommand for cutadapt for this new API?
